src/libraries/System.Console/src/System/Console.cs

```diff
@@ -421,8 +445,21 @@ public static void SetWindowPosition(int left, int top)
         }
 
         [SupportedOSPlatform("windows")]
+        [SupportedOSPlatform("linux")]
         public static void SetWindowSize(int width, int height)
         {
+            if (Console.IsOutputRedirected)
+            {
+                throw new IOException(SR.InvalidOperation_SetWindowSize);
```

Comment:
This isn't inside the lock, so is it possible the console could get redirected between this check and the call to the PAL, and if so would that matter?

Reply:
Thank you for the review.
As I see it from ConsolePal.Unix.cs, `Console.IsOurputRedirected` doesn't need lock and assumes it is fixed value.
https://github.com/dotnet/runtime/blob/7dad0f36e56161e1a92fc6660b8386d054439ff0/src/libraries/System.Console/src/System/ConsolePal.Unix.cs#L202-L203
https://github.com/dotnet/runtime/blob/7dad0f36e56161e1a92fc6660b8386d054439ff0/src/libraries/System.Console/src/System/ConsolePal.Unix.cs#L216-L219
https://github.com/dotnet/runtime/blob/7dad0f36e56161e1a92fc6660b8386d054439ff0/src/libraries/System.Console/src/System/ConsolePal.Unix.cs#L229-L232
https://github.com/dotnet/runtime/blob/7dad0f36e56161e1a92fc6660b8386d054439ff0/src/libraries/System.Console/src/System/ConsolePal.Unix.cs#L237-L238